Sources/BottomSheetView.swift

```diff
@@ -139,22 +140,30 @@ public final class BottomSheetView: UIView {
     /// Call this method e.g. when orientation change is detected.
     public func reset() {
         updateTargetOffsets()
-        animate(to: currentTargetOffset)
+
+        guard let superview = superview, let index = targetOffsets.firstIndex(where: { $0 < superview.frame.height }) else {
+            return
+        }
+
+        currentTargetOffsetIndex = index
+        transition(to: index)
     }
 
     /// Animates bottom sheet view to the given height.
     ///
     /// - Parameters:
     ///   - index: the index of the target height
     public func transition(to index: Int) {
-        guard index >= 0 && index < targetHeights.count else {
+        guard targetHeights.indices.contains(index) else {
             assertionFailure("Provided index is out of bounds of the array with target heights.")
             return
         }
-        guard let superview = superview else { return }
+        // Adds one to compansate for limit model at beginning
+        guard models.indices.contains(index + 1) else {
```

Comment:
Would this result in transition to incorrect index when `BottomSheetView` is not presented by a presentation controller and we don't add 0 for dismiss target height.

Reply:
Also, what happens to indices if you create the bottom sheet view with 0 height included? `BottomSheetView(contentView: contentView, targetHeights: [0, 100, 300]`)

Reply:
Could we perhaps do the same as before:
```swift
let offset = BottomSheetCalculator.offset(for: contentView, in: superview, height: targetHeights[index])
animate(to: offset)
```
```